### PR TITLE
Add a command revup commit

### PR DESCRIPTION
**Pull request by malcolm-leclair-skydio**
This command is an alias to "revup amend --insert"

Topic: revup_commit

### CODE REVIEW

**Comment by malcolm-leclair-skydio** [comment]
Reviews in this chain:
└**https://github.com/Skydio/revup/pull/52** **Add a command revup commit that aliases revup amend --insert**

**Comment by malcolm-leclair-skydio** [comment]
| # | head | base | diff | date | summary |
| - | - | - | - | - | - |
| 0 | [53b036bd](/Skydio/revup/commit/53b036bde7e26f6661e1adf04f1d2c519baf4d53) | [1ec4bcaa](/Skydio/revup/commit/1ec4bcaa6e1223048a02c0bf0f578655fee72f3b) | [diff](/Skydio/revup/compare/1ec4bcaa6e1223048a02c0bf0f578655fee72f3b..53b036bde7e26f6661e1adf04f1d2c519baf4d53) | Sep 20 12:46 PM |  1 file changed, 22 insertions(+), 11 deletions(-) |

File: revup/revup.py [hunk 1]
```diff
@@ -217,9 +217,10 @@ async def main() -> int:
         "cherry-pick",
         add_help=False,
     )
+    commit_parser = subparsers.add_parser("commit", add_help=False)
```
Comment (jerry-skydio) [inline revup/revup.py:220]: rather than add new parser, use aliases=["commit"]

Comment (jerry-skydio) [reply revup/revup.py:220]: although you might have to dig up the original invocation as so https://stackoverflow.com/questions/59493715/resolve-argparse-alias-back-to-the-original-command but I think this could be a useful addition to the RevupArgParser class

File: revup/revup.py [hunk 2]
```diff
@@ -319,9 +325,12 @@ async def main() -> int:
 
         return await cherry_pick.main(args=args, git_ctx=git_ctx)
 
-    elif args.cmd == "amend":
+    elif args.cmd in ["commit", "amend"]:
         from revup import amend
 
+        # Commit is an amend --insert
```
Comment (jerry-skydio) [inline revup/revup.py:331]: "commit" is an alias of "amend --insert"

**Comment by jerry-skydio** [comment]
as is this will result in a commit inserted 1 below the top commit rather than at the top. I can make a patch that fixes this since the original ordering of --insert was arbitrary anyway

File: revup/revup.py (hunk 1 above)
Comment (malcolm-leclair-skydio) [reply revup/revup.py]: Unless I'm misunderstanding something, I don't think that's necessary for this use case since we explicitly want to distinguish between the original "amend" and the alias "commit" to switch behavior

Comment (jerry-skydio) [reply revup/revup.py]: hmm i was assuming args.cmd would always end up as "amend" but looks like that's not the case. that's convenient then

File: revup/revup.py [hunk 3]
```diff
@@ -298,7 +298,13 @@ async def main() -> int:
 
     conf = await get_config()
 
-    for p in [revup_parser, amend_parser, cherry_pick_parser, restack_parser, upload_parser]:
+    for p in [
+        revup_parser,
+        amend_parser,
+        cherry_pick_parser,
+        restack_parser,
+        upload_parser,
```
Comment (jerry-skydio) [inline revup/revup.py:306]: undo this change as nothing was actually added to the list

**Review by jerry-skydio — APPROVED** [review]

**Comment by jerry-skydio** [comment]
can you squash all the commits in this pr into one? you can use revup amend ;)

**Comment by malcolm-leclair-skydio** [comment]
> can you squash all the commits in this pr into one? you can use revup amend ;)

Sure thing. Out of curiosity, why not squash merge from github?

**Comment by jerry-skydio** [comment]
> > can you squash all the commits in this pr into one? you can use revup amend ;)
> 
> Sure thing. Out of curiosity, why not squash merge from github?

there are some cases where we wouldn't want a blanket squash -- such as large changes that touch multiple area or are done in multiple stages.

the other reason is that the user can do a better job of cleaning up commit messages after the squash. github just slaps them all together
